Shape sorting game

dmclaughlin104 opened the pull request:
Created a basic gameplay loop for a shape sorting game. Wooden blocks appear in a random position in the middle of the screen, while their respective 'holes' appear in random positions along the sides. When all shapes are sorted in their correct holes, the shapes and holes are shuffled into new positions. Shapes glow yellow while being interacted with, and red when they are resting in the wrong spot. When they're in more or less the right spot, they will automatically move to centre of the hole and slot through it.

eoloughlin05 commented on Assets/Scripts/DragDropToDistance.cs:
```diff
@@ -0,0 +1,268 @@
+    private float moveSpeed = 0.9f;
+    [SerializeField] private string targetTag;
+
+    private bool isMoving = false; // Flag to track whether movement is in progress
```
Could we change this to 'isShapeMoving'?

eoloughlin05 commented on Assets/Scripts/DragDropToDistance.cs:
```diff
@@ -0,0 +1,268 @@
+    [SerializeField] private string targetTag;
+
+    private bool isMoving = false; // Flag to track whether movement is in progress
+    private bool isAtMiddle = false;//to check whether object has reached middle;
```
Could we change this to 'isShapeAtMiddleOfShapeSpace' or something indicating where the middle is?

eoloughlin05 commented on Assets/Scripts/DragDropToDistance.cs:
```diff
@@ -0,0 +1,268 @@
+    }
+
+    //move object backward and through hole
+    private void MoveBackWard()
```
We can change the method name of this to 'MoveObjectThroughHole' which would remove the comment line above.

eoloughlin05 commented on Assets/Scripts/DragDropToDistance.cs:
```diff
@@ -0,0 +1,268 @@
+    //methods to turn off/on the Rigid Body
+    private void TurnOffRB()
+    {
+        //GetComponent<Rigidbody>().useGravity = false;
```
This comment can be removed if no longer required.

eoloughlin05 commented on Assets/Scripts/DragDropToDistance.cs:
```diff
@@ -0,0 +1,268 @@
+    private void TurnOnRB()
+    {
+        //droping object - turn RB back on
+        //GetComponent<Rigidbody>().useGravity = true;
```
This comment can be removed if it is no longer required.

eoloughlin05 commented on Assets/Scripts/DragDropToDistance.cs:
```diff
@@ -0,0 +1,268 @@
+
+public class DragDropToDistance : MonoBehaviour
+{
+    //followed this tutorial for basic touchscreen dragging: https://www.youtube.com/watch?v=zo1dkYfIJVg
```
Can this line be removed?

eoloughlin05 commented on Assets/Scripts/DragDropToDistance.cs:
```diff
@@ -0,0 +1,268 @@
+    private bool isMoving = false; // Flag to track whether movement is in progress
+    private bool isAtMiddle = false;//to check whether object has reached middle;
+    private Transform targetTransform; // Reference to the transform of the target object
+
+
+
```
Lines 35 - 37 can be removed as it's just white space.

eoloughlin05 commented on Assets/Scripts/DragDropToDistance.cs:
```diff
@@ -0,0 +1,268 @@
+
+        //getting rigid body component
+        objectRB = GetComponent<Rigidbody>();
+
```
This line can be removed too - re. white space

eoloughlin05 commented on Assets/Scripts/DragDropToDistance.cs:
```diff
@@ -0,0 +1,268 @@
+        //declaring what should happen press interaction starts
+        press.performed += _ =>
+        {
+
```
This line can be removed - white space

eoloughlin05 commented on Assets/Scripts/DragDropToDistance.cs:
```diff
@@ -0,0 +1,268 @@
+            {
+                StartCoroutine(Drag());
+            }
+
```
This line can be removed - white space

eoloughlin05 commented on Assets/Scripts/RotatingCubesController.cs:
```diff
@@ -0,0 +1,86 @@
+
+public class RotatingCubesController : MonoBehaviour
+{
+
+    /*
+    [SerializeField] InputAction press;
+    private Vector2 touchStartPos;
+    //private bool isTouching = false;
+    [SerializeField] Camera playerCamera;
+    private Vector3 currentScreenPosition;
+    */
+
+    
```
These lines can be removed if the code isn't required anymore.

eoloughlin05 commented on Assets/Scripts/RotatingCubesController.cs:
```diff
@@ -0,0 +1,86 @@
+
+            // apply rotation
+            _rotation.y = -(_mouseOffset.x + _mouseOffset.y) * _sensitivity;
+            // apply rotation
+            //_rotation.x = -(_mouseOffset.x + _mouseOffset.y) * _sensitivity;
```
These lines can be removed if they're no longer required

eoloughlin05 commented on Assets/Scripts/RotatingCubesController.cs:
```diff
@@ -0,0 +1,86 @@
+        // rotating flag
+        _isRotating = false;
+    }
+
+    void HorizOrVert()
+    {
+
+    }
+
+
+    /*
+private void Awake()
+    {
+        //enabling interaction events
+        press.Enable();
+
+        press.performed += _ =>
+        {
+            
+        };
+
+    }
+    */
+    
```
These lines can be removed if they're not required.

eoloughlin05 commented on Assets/Scripts/ShapeSorting_GameManager.cs:
```diff
@@ -0,0 +1,160 @@
+
+public class ShapeSorting_GameManager : MonoBehaviour
+{
+
```
White space that can be removed

eoloughlin05 commented on Assets/Scripts/ShapeSorting_GameManager.cs:
```diff
@@ -0,0 +1,160 @@
+    public bool shapeReturning = false;
+    public int shapeToReturn;
+
+
```
Extra line can be removed - white space

eoloughlin05 commented on Assets/Scripts/ShapeSorting_PerimeterManager.cs:
```diff
@@ -0,0 +1,67 @@
+
+public class ShapeSorting_PerimeterManager : MonoBehaviour
+{
+
```
White space - can be removed

eoloughlin05 commented on Assets/Scripts/ShapeSorting_PerimeterManager.cs:
```diff
@@ -0,0 +1,67 @@
+
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
```
These lines can be removed as the Update method is empty and not in use.

eoloughlin05 reviewed (changes requested):
I have tested this out Damien and it looks great. Brilliant work!

Just a few comments; very minor changes such as changing a couple of the naming conventions so that it is clearer in the code what is happening. There are a couple of unused or empty methods that can be removed and some comment lines also which I have marked. The last thing is white space; where there are a lot of extra blank lines that can be removed. I have marked a few of these, but not them all, just to give you an example of where I'm talking about. If there's anything that doesn't make sense, please just let me know. I will review it again as soon as changes are made. Thanks!

dmclaughlin104 commented:
Thanks for this, Erin.

I've now updated the scripts to remove blank space and I've also deleted a few older scripts that weren't currently in use or which have been superseded (including the RotatingCubes script). I've also made the red wrong highlight more obvious now by making the whole front of the shape red.

I'm up for adding sound effects etc., but I think I they should be in-keeping with the new sounds that we've recorded for the colour-sorting game.

In terms of priorities, I'll probably need to look at bringing in the sound effects into the colour-sorting game first, and then I look at doing the same for the shape-sorting game.
